lib/jekyll-paginate-v2/generator/paginationModel.rb

```diff
@@ -267,6 +267,11 @@ def paginate(template, config, site_title, all_posts, all_tags, all_categories,
         indexPageExt = Utils.ensure_leading_dot(config['extension'])
         indexPageWithExt = indexPageName + indexPageExt
 
+        # In case there are no (visible) posts, generate the index file anyway
+        if total_pages == 0
+          total_pages = 1
+        end
```

Comment:
in one line.
```ruby
total_pages = 1 if total_pages.zero?
```

Reply:
Ok! Those were actually the first lines of Ruby I'd ever written... 😊